ndk/src/media/media_codec.rs

```diff
@@ -26,7 +26,7 @@ pub enum MediaCodecDirection {
 /// A native [`AMediaCodec *`]
 ///
 /// [`AMediaCodec *`]: https://developer.android.com/ndk/reference/group/media#amediacodec
-#[derive(Debug)]
```

Comment:
No clue about this API, but if there aren't multiple `MediaCodec` with  the same `AMediaCodec`, then you could implement `Eq`/`Hash` by hand over just that and not use the callback.

Reply:
`AMediaCodec` doesn't implement refcounting semantics so there should always be exactly one `MediaCodec` for a given `AMediaCodec` pointer, so this is definitely possible here.

That is also why we can lifetime the callbacks within the wrapper object in the first place (as we deduced is possible in for example Thermal Manager as well, but not in Performance Manager which is a global `static`).